test/common/index.js

```diff
@@ -501,6 +501,12 @@ exports.mustCallAtLeast = function(fn, minimum) {
   return _mustCallInner(fn, minimum, 'minimum');
 };
 
+exports.mustCallAsync = function(fn, exact) {
+  return exports.mustCall((...args) => {
+    fn(...args).then(exports.mustCall());
+  }, exact);
```

Comment:
Should the returned function return a promise as well, so that one could chain the functions returned by mustCallAsync?

Reply:
@TimothyGu yup, done!